### DISCUSSION TITLE
Web assembly

### DISCUSSION THREAD

**Discussion by amirhossein-fzl** [vlang/v#18616] [Ideas]
Please add web assembly to V. If you build a framework like Blazor in C# along with web assembly, it will be great! This framework is designed in such a way that the front-end and back-end of the application are connected with sockets in real time. But in development, getting data from the backend is as simple as writing a variable or calling a method!

**Comment by ArtemkaKun** [comment]
Hi,

Afaik there is not `native` way to build an web assembly in V for now. But you can do that with tools that builds C code into WASM (e. g. emscripten). You can check the `wasm-backend` chat on the V Discord channel.

About Blazor - this is not about WASM, but about SignalR. Blazor WASM - this is a "standalone" app, and ` the front-end and back-end of the application are connected with sockets in real time` - this is the Blazor Server. Afaik SignalR is based on websockets (as you mentioned), and V already has websockets, so this is possible to port SignalR to V or implement another way to have SignalR behavior in V.
